### PR TITLE
new style linear_interpolate function

### PR DESCRIPTION
**Pull request by kmaehashi**
This PR implements new style `linear_interpolate` function.

### CODE REVIEW

**Comment by kmaehashi** [comment]
Removed retry decorator, confirmed 10000 repeats.

File: chainer/functions/math/linear_interpolate.py [hunk 1]
```diff
@@ -52,6 +62,15 @@ def backward_gpu(self, inputs, grads):
...
         )(p, x, y, g)
 
+    def backward(self, indexes, grad_outputs):
+        p, x, y, g = self.get_retained_inputs()
+        g0, g1, g2 = grad_outputs
+        ggp = g * (g1 - g2)
+        ggx = g * g0
+        ggy = - ggx
+        gg = (x - y) * g0 + p * g1 + (1 - p) * g2
+        return ggp, ggx, ggy, gg
```
Comment (takagi) [inline chainer/functions/math/linear_interpolate.py]: `gp`, `gx` and `gy` would be more natural than `ggp`, `ggx` and `ggy`?

File: tests/chainer_tests/functions_tests/math_tests/test_linear_interpolate.py [hunk 2]
```diff
@@ -22,13 +21,19 @@ def setUp(self):
         self.x = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
         self.y = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
         self.g = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
+        self.gp = numpy.random.uniform(0, 1, self.shape).astype(self.dtype)
```
Comment (takagi) [inline tests/chainer_tests/functions_tests/math_tests/test_linear_interpolate.py]: `self.ggp`?

File: tests/chainer_tests/functions_tests/math_tests/test_linear_interpolate.py [hunk 3]
```diff
@@ -22,13 +21,19 @@ def setUp(self):
...
+        self.gx = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
```
Comment (takagi) [inline tests/chainer_tests/functions_tests/math_tests/test_linear_interpolate.py]: `self.ggx`?

File: tests/chainer_tests/functions_tests/math_tests/test_linear_interpolate.py [hunk 4]
```diff
@@ -22,13 +21,19 @@ def setUp(self):
...
+        self.gy = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
```
Comment (takagi) [inline tests/chainer_tests/functions_tests/math_tests/test_linear_interpolate.py]: `self.ggy`?

**Comment by takagi** [comment]
Would you rebase on master, which would improve test stability by changes since the last commit?

**Comment by kmaehashi** [comment]
Thanks for review, fixed accordingly.

**Comment by takagi** [comment]
LGTM, will merge after CI pass!

**Pull request merged**